File: app/src/main/java/com/afollestad/photoaffix/views/LineView.java
package com.afollestad.photoaffix.views;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.support.annotation.ColorInt;
import android.util.AttributeSet;
import android.util.TypedValue;
import android.view.View;

/** @author Aidan Follestad (afollestad) */
public class LineView extends View {

  private Paint paint;

  public LineView(Context context) {
    super(context);
    init();
  }

  public LineView(Context context, AttributeSet attrs) {
    super(context, attrs);
    init();
  }

  public LineView(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
    init();
  }

  private void init() {
    setWillNotDraw(false);
    paint = new Paint();
    paint.setAntiAlias(true);
    paint.setColor(Color.BLACK);
    paint.setStyle(Paint.Style.FILL_AND_STROKE);
    if (isInEditMode()) setWidth(8);
  }

  public void setWidth(int width) {
    width =
        (int)
            TypedValue.applyDimension(
                TypedValue.COMPLEX_UNIT_DIP, width, getResources().getDisplayMetrics());
    paint.setStrokeWidth(width);
    invalidate();
  }

  public void setColor(@ColorInt int color) {
    paint.setColor(color);
    invalidate();
  }

  @Override
  protected void onDraw(Canvas canvas) {
    super.onDraw(canvas);
    if (getTag() != null && getTag().equals("1")) {
      canvas.drawLine(
          0, (getMeasuredHeight() / 2), getMeasuredWidth(), (getMeasuredHeight() / 2), paint);
    } else {
      canvas.drawLine(
          (getMeasuredWidth() / 2), 0, (getMeasuredWidth() / 2), getMeasuredHeight(), paint);
    }
  }
}

File: app/src/main/java/com/afollestad/photoaffix/views/LineView.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.photoaffix.views

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import androidx.annotation.ColorInt
import android.util.AttributeSet
import android.util.TypedValue.COMPLEX_UNIT_DIP
import android.util.TypedValue.applyDimension
import android.view.View

/** @author Aidan Follestad (afollestad) */
class LineView(
  context: Context,
  attrs: AttributeSet? = null
) : View(context, attrs) {

  private var paint: Paint

  init {
    setWillNotDraw(false)
    paint = Paint().apply {
      isAntiAlias = true
      color = Color.BLACK
      style = Paint.Style.FILL_AND_STROKE
    }
    if (isInEditMode) {
      width = 8
    }
  }

  fun setWidth(width: Int) {
    val newWidth = applyDimension(COMPLEX_UNIT_DIP, width.toFloat(), resources.displayMetrics)
        .toInt()
    paint.strokeWidth = newWidth.toFloat()
    invalidate()
  }

  fun setColor(@ColorInt color: Int) {
    paint.color = color
    invalidate()
  }

  override fun onDraw(canvas: Canvas) {
    super.onDraw(canvas)
    if (tag != null && tag == "1") {
      canvas.drawLine(
          0f,
          (measuredHeight / 2).toFloat(),
          measuredWidth.toFloat(),
          (measuredHeight / 2).toFloat(),
          paint
      )
    } else {
      canvas.drawLine(
          (measuredWidth / 2).toFloat(),
          0f,
          (measuredWidth / 2).toFloat(),
          measuredHeight.toFloat(),
          paint
      )
    }
  }
}
